benchmarks/perf-tool/okpt/test/steps/steps.py

```diff
@@ -414,6 +483,115 @@ def _get_measures(self) -> List[str]:
 
         return measures
 
+    @abstractmethod
+    def get_body(self, vec):
+        pass
+
+
+class QueryStep(BaseQueryStep):
+    """See base class."""
+
+    label = 'query'
+
+    def __init__(self, step_config: StepConfig):
+        super().__init__(step_config)
+        self.neighbors = parse_dataset(self.neighbors_format, self.neighbors_path,
+                                       Context.NEIGHBORS)
+        self.implicit_config = step_config.implicit_config
+
+    def get_body(self, vec):
+        return {
+            'size': self.k,
+            'query': {
+                'knn': {
+                    self.field_name: {
+                        'vector': vec,
+                        'k': self.k
+                    }
+                }
+            }
+        }
+
+
+class QueryWithFilterStep(BaseQueryStep):
+    """See base class."""
+
+    label = 'query_with_filter'
+
+    def __init__(self, step_config: StepConfig):
+        super().__init__(step_config)
+
+        neighbors_dataset = parse_string_param('neighbors_dataset',
+                                               step_config.config, {}, None)
+
+        self.neighbors = parse_dataset(self.neighbors_format, self.neighbors_path,
```

Comment:
Why is this CUSTOM and not NEIGHBORS?

Reply:
we do have a dataset for each filter, instead of making it multiple files with same dataset name I do one file with multiple datasets for each filter. It makes it a bit easy if there are many filters, say for lucene benchmarking I used 5.